### PR TITLE
DOC: Update to docstring of DataFrame(dtype) (#14764)

### LINKED ISSUE #14764: DOC: pd.DataFrame(dtype) arg cannot be list, dict, Series. And None will infer wider type than necessary.
#### Code Sample, a copy-pastable example if possible

```python

# This is a DOCbug simply to document what pd.DataFrame(dtype) currently does.
# It is non-obvious to new users, gives non-obvious error messages, and also behaves differently to read_csv(dtype)

# a) Leaving dtype=None in constructor will infer a wider type than necessary
df_cols = {'year':np.int32, 'month':np.int8}
df = pd.DataFrame(columns=df_cols.keys(), dtype=None, index=range(10), data=-1)
>>> df.dtypes
month    int64
year     int64

# b) The doc doesn't explicitly say a list/dict/Series/array-like is not allowed (and if you
# pass in one the error is not very friendly). Also behaves differently to read_csv(dtype)
df = pd.DataFrame(columns=df_cols.keys(), dtype=np.int32, index=range(10), data=-1)
# df.dtypes shows they're all np.int32
# Fix up dtypes after declaration
for col,coltype in df_cols.items():
    df[col] = df[col].astype(coltype) 

```
#### Problem description

The DataFrame() doc doesn't explicitly say a list/dict/Series/array-like is not allowed (and if you pass in one the error is not very friendly). Also behaves differently to read_csv(dtype).
Leaving dtype=None in constructor will infer a wider type than necessary.
So in general you either set dtype=widest_necessary_type, or dtype=None and then manually fix them up after declaration, by casting with `astype()`

#### Output of ``pd.show_versions()``

<details>
python: 2.7.10.final.0
python-bits: 64
machine: x86_64
processor: i386
byteorder: little
pandas: 0.19.1
numpy: 1.11.2
scipy: 0.18.1
</details>

### PR DESCRIPTION
**Pull request by VincentLa**
- [ ] closes #14764 
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -241,17 +241,52 @@ class DataFrame(NDFrame):
         Column labels to use for resulting frame. Will default to
         np.arange(n) if no column labels are provided
     dtype : dtype, default None
-        Data type to force, otherwise infer
+        Data type to force. Only a single dtype is allowed. If None, infer
```
Comment (jreback) [inline pandas/core/frame.py]: this is not right. you can pass a dict of dtypes.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -241,17 +241,52 @@ class DataFrame(NDFrame):
...
+    col1    int8
+    col2    int8
+    dtype: object
+
+    Constructing DataFrame from numpy ndarray:
+
+    >>> df2 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
+    >>> df2
+        a           b           c           d           e
```
Comment (jreback) [inline pandas/core/frame.py]: these are not lined up, something odd going on.

Comment (VincentLa) [inline pandas/core/frame.py]: I think this is ok? It comes out looking like this when I generated
screen shot 2017-05-24 at 4 18 57 pm

File: pandas/core/frame.py (hunk 1 above)
Comment (VincentLa) [inline pandas/core/frame.py]: Oh interesting. How? This does not work for me:
```
d = {'col1': [1, 2], 'col2': [3, 4]}
df = pd.DataFrame(data=d, dtype={'col1': np.int64, 'col2': np.int64})
```

File: pandas/core/frame.py (hunk 2 above)
Comment (jreback) [inline pandas/core/frame.py]: can you line up on the decimals.

Comment (VincentLa) [inline pandas/core/frame.py]: sure I'll just change the example to something easier with integers to take up less space anyway.

File: pandas/core/frame.py (hunk 1 above)
Comment (VincentLa) [inline pandas/core/frame.py]: According to https://github.com/pandas-dev/pandas/issues/4464 and https://github.com/pandas-dev/pandas/issues/9133 it seems like this is not yet implemented.

Comment (jreback) [inline pandas/core/frame.py]: https://github.com/pandas-dev/pandas/pull/13375 implemented his for astype
hasn't been implemented for construction

**Review by jorisvandenbossche — APPROVED** [review]
Looks good! (I did a small cosmetic edit, see latest commit)

**Comment by jorisvandenbossche** [comment]
@VincentLa Thanks!

**Pull request merged**